pandas/core/algorithms.py

```diff
@@ -911,10 +911,15 @@ def select_n_frame(frame, columns, n, method, keep):
     if not is_list_like(columns):
         columns = [columns]
     columns = list(columns)
+    tmp = Series(frame.index)
+    frame.reset_index(inplace=True, drop=True)
     ser = getattr(frame[columns[0]], method)(n, keep=keep)
     if isinstance(ser, Series):
         ser = ser.to_frame()
```

Comment:
Hmm, not sure we can do that @jreback, take this for example
```python
df = pd.DataFrame(dict(
    a=[2, 2, 1, 1, 1],
    b=[4, 3, 8, 9, 5],
))
result = df.nsmallest(2, ['a', 'b'])
```
If we do nsmallest on col `a` and then sort on cols `a` and `b`
we'll get 

```
   a  c
2  1  8
3  1  9
```
when the desired result is
```
   a  c
4  1  5
2  1  8
```